From a162b428531b77fc155354ea00494b3395da4824 Mon Sep 17 00:00:00 2001
From: Liu Ying <Ying.Liu@freescale.com>
Date: Tue, 3 Mar 2015 11:36:53 +0800
Subject: [PATCH 0210/1221] MLK-10436-03: ARM: imx: Add PRG0/1 clocks as
 shared gate clocks for i.MX6QP

There are two PRG engines embedded in i.MX6Q R2.
The PRG0/1 axi gate clocks are derived from the ipu1/2_podf clocks respectively.
The PRG0/1 apb gate clocks are derived from the ipg clock.
The gating control bits are shared by the PRG0/1 axi/apb gate clocks.
This patch adds the gate clocks support as shared gate clocks.

Signed-off-by: Liu Ying <Ying.Liu@freescale.com>
Signed-off-by: Bai Ping <b51503@freescale.com>
[zou:Original patch taken from
git.freescale.com/imx/fsl-arm-yocto-bsp.git-b imx-3.14.52-1.1.0_ga]
Signed-off-by: zou cao <cao.zou@windriver.com>
---
 arch/arm/mach-imx/clk-imx6q.c             |    6 ++++++
 include/dt-bindings/clock/imx6qdl-clock.h |    6 +++++-
 2 files changed, 11 insertions(+), 1 deletions(-)

diff --git a/arch/arm/mach-imx/clk-imx6q.c b/arch/arm/mach-imx/clk-imx6q.c
index e2fb7dc..cc82b95 100644
--- a/arch/arm/mach-imx/clk-imx6q.c
+++ b/arch/arm/mach-imx/clk-imx6q.c
@@ -134,6 +134,8 @@ static unsigned int share_count_ssi1;
 static unsigned int share_count_ssi2;
 static unsigned int share_count_ssi3;
 static unsigned int share_count_spdif;
+static unsigned int share_count_prg0;
+static unsigned int share_count_prg1;
 
 static void __init imx6q_clocks_init(struct device_node *ccm_node)
 {
@@ -539,6 +541,10 @@ static void __init imx6q_clocks_init(struct device_node *ccm_node)
 		clk[IMX6QDL_CLK_PRE1] = imx_clk_gate2("pre1",          "gpu3d_axi",         base + 0x80, 18);
 		clk[IMX6QDL_CLK_PRE2] = imx_clk_gate2("pre2",          "gpu3d_axi",         base + 0x80, 20);
 		clk[IMX6QDL_CLK_PRE3] = imx_clk_gate2("pre3",          "gpu3d_axi",         base + 0x80, 22);
+		clk[IMX6QDL_CLK_PRG0_AXI] = imx_clk_gate2_shared("prg0_axi", "ipu1_podf",   base + 0x80, 24, &share_count_prg0);
+		clk[IMX6QDL_CLK_PRG1_AXI] = imx_clk_gate2_shared("prg1_axi", "ipu2_podf",   base + 0x80, 26, &share_count_prg1);
+		clk[IMX6QDL_CLK_PRG0_APB] = imx_clk_gate2_shared("prg0_apb", "ipg",         base + 0x80, 24, &share_count_prg0);
+		clk[IMX6QDL_CLK_PRG1_APB] = imx_clk_gate2_shared("prg1_apb", "ipg",         base + 0x80, 26, &share_count_prg1);
 	}
 	clk[IMX6QDL_CLK_CKO1]         = imx_clk_gate("cko1",           "cko1_podf",         base + 0x60, 7);
 	clk[IMX6QDL_CLK_CKO2]         = imx_clk_gate("cko2",           "cko2_podf",         base + 0x60, 24);
diff --git a/include/dt-bindings/clock/imx6qdl-clock.h b/include/dt-bindings/clock/imx6qdl-clock.h
index e7a9ad7..4dfa7e7 100644
--- a/include/dt-bindings/clock/imx6qdl-clock.h
+++ b/include/dt-bindings/clock/imx6qdl-clock.h
@@ -265,6 +265,10 @@
 #define IMX6QDL_CLK_PRE1			256
 #define IMX6QDL_CLK_PRE2			257
 #define IMX6QDL_CLK_PRE3			258
-#define IMX6QDL_CLK_END				259
+#define IMX6QDL_CLK_PRG0_AXI			259
+#define IMX6QDL_CLK_PRG1_AXI			260
+#define IMX6QDL_CLK_PRG0_APB			261
+#define IMX6QDL_CLK_PRG1_APB			262
+#define IMX6QDL_CLK_END				263
  
 #endif /* __DT_BINDINGS_CLOCK_IMX6QDL_H */
-- 
1.7.5.4

